_episodes/16-file-formats.md

```diff
@@ -0,0 +1,68 @@
+---
+title: "File Formats"
+teaching: 10
+exercises: 0
+questions:
+- "How can I allow type-checking of input and output files?"
+objectives:
+- "Learn how to unambiguously specify the format of `File` objects."
+keypoints:
+- "You can document the expected format of input and output `File`s."
+- "Once your tool is mature, we recommend specifying formats by referencing
+existing ontologies e.g. EDAM."
+---
+Tools and workflows can take `File` types as input and produce them as output.
+We also recommend indicating the format for `File` types. This helps document
+for others how to use your tool while allowing you to do some simple
+type-checking when creating parameter files.
+
+For file formats, we recommend referencing existing ontologies (like EDAM in
+our example), reference a local ontology for your institution, or do not add
+a file format initially for quick development before sharing your tool with
+others.
+
+Note that for added value `cwltool` can do some basic reasoning based on file
+formats and warn you if there seem to be some obvious mismatches.
+
+*metadata_example.cwl*
```

Comment:
seems like we might need a slightly more complicated example than `wc -l` on a BAM file. Any preference? 

my concern is that we shouldn't encourage anyone to mic the IANA and EDAM ontologies unless they have to. I'm not familiar enough with either one to know what's unique to one or the other.

I'll do some digging...